99 email validation

Nkoli opened the pull request:
Fixes #99

Commit:
create email validation

Commit:
Add validation for email field

Commit:
Delete validate function

Commit:
add validation check for email

Commit:
comment out failing test

kosyfrances commented on app/serializers.py:
```diff
@@ -81,7 +85,7 @@ def update(self, instance, validated_data):
 
         instance.username = validated_data.get("username", instance.username)
-        instance.email = validated_data.get("email", instance.email)
+        # instance.email = validated_data.get("email", instance.email)
```
You need add email to "instance".

kosyfrances commented on app/tests/test_user.py:
```diff
@@ -51,22 +51,22 @@ def test_create_new_user(self):
+    #             "username": "testusertwo",
+    #             "skills": [self.skill2.id],
+    #             "timezone": "Africa/Cairo",
+    #         },
```
When doing a put or edit, you also need to ensure that required fields like email are sent along

Commit:
Add email field to profile update test, add email to update serializer instance

Commit:
edit help text

Commit:
create function for retrieval of private/hidden emails

Commit:
add view function for retrieving private verified email from the list returned by github

Commit:
add test for email retrieval

Commit:
append the 'accept' header

Commit:
add line spacing

Commit:
reorder mock return values

Commit:
edit username field

Commit:
edit email's return value to its correct type

Commit:
edit gitlab username field

Commit:
replace duplicate .get() function with the variable name to fix duplicate key integrity error

Commit:
create user model clean method to validate fields.
- also override default save method so that it implements the customized clean() method

Commit:
add email field and value to mentor test object

kosyfrances reviewed (commented):
LGTM but add a test checking for when the email is empty in github oauth

Commit:
edit docstring

Commit:
add test to check that user object cannot be created without adding a valid email address
